src/middlewares/Fetch.ts

```diff
@@ -15,8 +17,11 @@ export type FetchResponse<T> = {
     statusText: string;
     type: string;
     url: string;
+    cacheUsed?: boolean;
```

Comment:
thats fine, do we do so with a whole new typedef in the Cache.ts as the typedef `FetchRequest` from `Fetch` doesnt seem to be extendable here to add two new things